test/enums.jl

```diff
@@ -179,6 +179,20 @@ end
 @enum HashEnum2 Enum2_a=1
 @test hash(Enum1_a) != hash(Enum2_a)
 
+# PR #49777: Check that `Base.hash` can be specialized by the user without
+# overwriting a method definition.
+let
+    redir_err = "redirect_stderr(stdout)"
+    exename = Base.julia_cmd()
+    script = """
+        $redir_err
+        @enum MyEnum A = 1
+        Base.hash(x::MyEnum, h::UInt) = h
+        """
+    warning_str = read(`$exename --warn-overwrite=yes --startup-file=no -e $script`, String)
```

Comment:
Starting a whole new process for this feels excessive. Maybe just checking that there is no hash method defined for a specific enum after it is created is enough?

Reply:
thanks, fixed